File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    num_list = [True] * (n + 1)
    num_list[0] = False
    num_list[1] = False
    
    for num in range(2, int(n**0.5) + 1):
        if num_list[num] == True:
            for i in range(num * num, n + 1, num):
                num_list[i] = False
    
    return num_list.count(True)

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const N = Number(input[0]);
const numbers = input[1].split(' ').map(Number);
let count = 0;

for (let num of numbers) {
    if (num < 2) {
        continue;
    }
    
    let isPrime = true;
    for (let i = 2; i < num; i++) {
        if (num % i === 0) {
            isPrime = false;
            break;
        }
    }
    
    if (isPrime) {
        count++;
    }
}

console.log(count);